### PR TITLE
Nistats update

### PR DESCRIPTION
**Pull request by swaythe**
@bthirion CircleCI passes! I updated the nistats references in the examples run by CircleCI to use the nilearn.stats functions instead, and circleci now passes. It's time nistats was merged into nilearn! :~) 

This PR doesn't have to be merged as is, I created it more for proof-of-concept; there is some ugly commenting of code in the examples that should be removed before merging.

### CODE REVIEW

File: examples/pipelining/nistats_glm_fsl_feeds_fmri.py [hunk 1]
```diff
@@ -48,11 +48,11 @@
...
 drift_model = 'Cosine'
 hrf_model = 'spm + derivative'
-design_matrix = make_design_matrix(frame_times=frametimes,
-                                   paradigm=paradigm,
+design_matrix = make_first_level_design_matrix(frame_times=frametimes,
+                                   events=paradigm,
                                    hrf_model=hrf_model,
                                    drift_model=drift_model,
-                                   period_cut=hfcut)
+                                   high_pass=hfcut)
```
Comment (bthirion) [inline examples/pipelining/nistats_glm_fsl_feeds_fmri.py:55]: But this value should be about 0.01

File: examples/pipelining/nistats_glm_fsl_feeds_fmri.py [hunk 2]
```diff
@@ -111,32 +111,28 @@
...
+    #     map_path = os.path.join(
+    #         map_dir, '%s.nii.gz' % contrast_id)
+    #     nibabel.save(out_map, map_path)
+    #
+    #     if dtype == "z":
+    #         z_maps[contrast_id] = map_path
+    #
+    #     print("\t\t%s map: %s" % (dtype, map_path))
+    #
+    # print
```
Comment (bthirion) [inline examples/pipelining/nistats_glm_fsl_feeds_fmri.py:135]: Please remove this

File: examples/pipelining/nistats_glm_fsl_feeds_fmri.py [hunk 3]
```diff
@@ -166,6 +161,6 @@
     )
 
 # shutdown main report page
-ProgressReport().finish_dir(output_dir)
+# ProgressReport().finish_dir(output_dir)
```
Comment (bthirion) [inline examples/pipelining/nistats_glm_fsl_feeds_fmri.py:164]: remove commented lines

File: pypreprocess/reporting/preproc_reporter.py [hunk 4]
```diff
@@ -744,6 +744,7 @@ def generate_tsdiffana_thumbnail(image_files, sessions, subject_id,
                         for i in range(len(figures))]
     for fig, output_filename in zip(figures, output_filenames):
         fig.savefig(output_filename, bbox_inches="tight", dpi=200)
+        # fig.savefig(output_filename, format='pdf')
```
Comment (bthirion) [inline pypreprocess/reporting/preproc_reporter.py:747]: same

File: pypreprocess/reporting/preproc_reporter.py [hunk 5]
```diff
@@ -792,6 +793,7 @@ def generate_realignment_thumbnails(
         pl.axvline(aux + l, linestyle="--", c="k")
         aux += l
     pl.savefig(rp_plot, bbox_inches="tight", dpi=200)
+    # pl.savefig(rp_plot, format='pdf')
```
Comment (bthirion) [inline pypreprocess/reporting/preproc_reporter.py:796]: +1

File: pypreprocess/reporting/preproc_reporter.py [hunk 6]
```diff
@@ -879,6 +881,7 @@ def _get_time_series_from_voxel(x, voxel):
         pl.xlabel('time (TR)')
 
         pl.savefig(output_filename, bbox_inches="tight", dpi=200)
+        # pl.savefig(output_filename, format='pdf')
```
Comment (bthirion) [inline pypreprocess/reporting/preproc_reporter.py:884]: +1

**Comment by swaythe** [comment]
I removed the comments and changed the high_pass to 1/hfcut instead of hfcut. Now the example fails, and my guess is it's somewhere in fitting the GLM. I'm looking into it, but do you have an intuition?

**Comment by bthirion** [comment]
The error comes from an old bug in Pypreprocess glm reporter, that we should replace with nilearn. Let me see if we can simply modify the example for the time being and replace the reporter later in the future.

**Comment by bthirion** [comment]
I have a hard time reproducing this. For the time being, could you simply skip the report generation in the example ?
thx

**Comment by swaythe** [comment]
Done. CircleCI still passes. :~)

File: examples/pipelining/nipype_preproc_spm_multimodal_faces.py [hunk 7]
```diff
@@ -56,6 +56,7 @@
 drift_model = 'Cosine'
 hrf_model = 'spm + derivative'
 hfcut = 128.
+hfcut = 1./hfcut
```
Comment (bthirion) [inline examples/pipelining/nipype_preproc_spm_multimodal_faces.py:59]: Please directly write `hfcut = 1. /128`

File: pypreprocess/reporting/glm_reporter.py [hunk 8]
```diff
@@ -443,7 +443,7 @@ def group_one_sample_t_test(masks, effects_maps, contrasts, output_dir,
     for contrast_id in contrasts:
         print("\tcontrast id: %s" % contrast_id)
 
-        # effects maps will be the input to the second level GLM
+        # effects maps will be the input to the second level GLMfirmfirmfirmfirm
```
Comment (bthirion) [inline pypreprocess/reporting/glm_reporter.py:446]: typo

**Review by bthirion — COMMENTED** [review]
We're almost there ! Thx,

**Comment by swaythe** [comment]
Done.

File: examples/easy_start/nipype_preproc_spm_auditory.py [hunk 9]
```diff
@@ -76,8 +78,8 @@
 
 # fit GLM
 print('\r\nFitting a GLM (this takes time) ..')
-fmri_glm = FirstLevelGLM(noise_model='ar1', standardize=False).fit(
-    [nibabel.concat_images(subject_data.func[0])], design_matrix)
+fmri_glm = FirstLevelModel(noise_model='ar1', standardize=False, t_r = tr).fit(
```
Comment (bthirion) [inline examples/easy_start/nipype_preproc_spm_auditory.py:81]:
```suggestion
fmri_glm = FirstLevelModel(noise_model='ar1', standardize=False, t_r=tr).fit(
```

File: examples/pipelining/nipype_preproc_spm_multimodal_faces.py [hunk 10]
```diff
@@ -55,7 +55,7 @@
 tr = 2.
 drift_model = 'Cosine'
 hrf_model = 'spm + derivative'
-hfcut = 128.
+hfcut = 1./128
```
Comment (bthirion) [inline examples/pipelining/nipype_preproc_spm_multimodal_faces.py:58]:
```suggestion
hfcut = 1. / 128
```

**Review by bthirion — COMMENTED** [review]
OK two mini-glitches to fix and we merge !

**Comment by bthirion** [comment]
Note that we'll remove the glm reporter from pypreprocess and rely on that of Nilearn.

**Comment by swaythe** [comment]
That makes sense, there's not much point in fixing pypreprocess reporting.

**Review by bthirion — APPROVED** [review]

**Comment by bthirion** [comment]
Thx for this !

**Pull request merged**